pkg/db/canary.go

```diff
@@ -121,22 +139,43 @@ func GetTransformedCheckIDs(canaryID string) ([]string, error) {
 	return ids, err
 }
 
-func UpdateChecksStatus(ids []string, status models.CheckHealthStatus) error {
+func RemoveTransformedChecks(ids []string, status models.CheckHealthStatus) error {
 	if len(ids) == 0 {
 		return nil
 	}
-	if !utils.Contains(models.CheckHealthStatuses, status) {
-		return fmt.Errorf("invalid check health status: %s", status)
+	updates := map[string]any{
+		"deleted_at": gorm.Expr("NOW()"),
+	}
+	if status != "" {
+		if !utils.Contains(models.CheckHealthStatuses, status) {
+			return fmt.Errorf("invalid check health status: %s", status)
+		}
+		updates["status"] = status
 	}
 	return Gorm.Table("checks").
```

Comment:
I think we want to insert a new check rather than update the old checks

Reply:
Do you mean check_status ?
Ie. Insert new check status and mark the existing check as deleted ?